src/index.js

```diff
@@ -10,17 +11,20 @@ var expr = module.exports = {
     },
   compiler: function(args, opt) {
       args = args.slice();
+      args.unshift('fn');
       var generator = codegen(opt),
           len = args.length,
           compile = function(str) {
             var value = generator(expr.parse(str));
             args[len] = '"use strict"; return (' + value.code + ');';
-            value.fn = Function.apply(null, args);
+            var generatedFn = Function.apply(null, args);
+            value.fn = generatedFn.bind(null, generator.functionDefinitions);
```

Comment:
Do you know if there are any performance/optimization hits associated with using the results of `bind` vs. a more "immediate" function? (These functions are often invoked in an inner loop over potentially thousands of data points.)